lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -612,14 +627,20 @@ def do_3d_projection(self):
         return np.min(vzs) if vzs.size else np.nan
 
     def _maybe_depth_shade_and_sort_colors(self, color_array):
-        color_array = (
-            _zalpha(color_array, self._vzs)
-            if self._vzs is not None and self._depthshade
-            else color_array
-        )
+        # Adjust the color_array alpha values if point depths are defined
+        # and depth shading is active
+        if self._vzs is not None and self._depthshade:
+            color_array = _zalpha(color_array, self._vzs, _dscl=self._dscl)
+
+        # Adjust the order of the color_array using the _z_markers_idx,
+        # which has been sorted by z-depth
         if len(color_array) > 1:
             color_array = color_array[self._z_markers_idx]
-        return mcolors.to_rgba_array(color_array, self._alpha)
+
+        # The correct alpha is already accounted for in the color_array
+        # Passing the self._alpha value in overwrites the automatically
+        # adjusted alpha
```

Comment:
This comment is about code that previously existed, which is confusing; please reword so it's about the current code.

Reply:
Will remove since no comment was previously needed. I was just noting why it was changed.